types/nationalAvalancheCenter/schemas.ts

```diff
@@ -16,6 +17,11 @@ import {
   Units,
 } from './enums';
 
+const datePipeline = z
+  .string()
+  .transform(s => fixMalformedISO8601DateString(s))
+  .transform(s => new Date(s));
```

Comment:
1. not married to the name `datePipeline`, LMK if you have suggestions
2. probably marginally faster to pass a single function to `.transform`, I was trying to be super-explicit here but I can be faster instead